llvm/include/llvm/IR/IntrinsicsAMDGPU.td

```diff
@@ -1927,6 +1927,13 @@ def int_amdgcn_inverse_ballot :
   Intrinsic<[llvm_i1_ty], [llvm_anyint_ty],
             [IntrNoMem, IntrWillReturn, IntrNoCallback, IntrNoFree]>;
 
+// Since there is no good VALU equivalent of this instruction, we mark it as convergent.
+// This prevents tranformations from uniform input to divergent input.
+// If the input is in a VGPR, we insert a v_readfirstlane.
```

Comment:
Please add a further comment on why we don't need a waterfall loop here.

Reply:
I think this should just say: "Lowers to S_BITREPLICATE_B64_B32. The argument must be uniform; otherwise, the result is undefined."

Reply:
Basically, prefer documenting things in terms of the contract between the frontend and the backend instead of explaining / restating the current implementation.

Reply:
> Basically, prefer documenting things in terms of the contract between the frontend and the backend instead of explaining / restating the current implementation.

I adjusted the comment. That is a good way of thinking about documentation.